crates/runtime/src/dataaccelerator.rs

```diff
@@ -201,10 +210,14 @@ impl AcceleratorExternalTableBuilder {
         }
 
         if !self.indexes.is_empty() {
-            let indexes_option_str = Acceleration::indexes_to_option_string(&self.indexes);
+            let indexes_option_str = Acceleration::hashmap_to_option_string(&self.indexes);
             options.insert("indexes".to_string(), indexes_option_str);
         }
 
+        if let Some(on_conflict) = self.on_conflict {
+            options.insert("on_conflict".to_string(), on_conflict.to_string());
```

Comment:
This is where we serialize the `on_conflict` option to string to be passed to the TableProviderFactory.